tools/benchmarks/run_service_api.ipynb

```diff
@@ -0,0 +1,184 @@
+    "        name='pipeline-' + random_suffix(),\n",
+    "        url=api_url)\n",
+    "    pipeline = client.pipelines.create_pipeline(body=api_pipeline)\n",
+    "    default_version_id = pipeline.default_version.id\n",
+    "\n",
+    "    # Create an experiment.\n",
+    "    experiment_name = 'experiment-' + random_suffix()\n",
+    "    experiment = client.experiments.create_experiment(body={'name' : experiment_name})\n",
+    "    experiment_id = experiment.id\n",
+    "    \n",
+    "    # Measure create run latency. Note this time is the roundrip latency of CreateRun method. The actual run is \n",
+    "    # not finished when client side gets the CreateRun response. Run duration will be measured below when run is \n",
+    "    # actually finished.\n",
+    "    created_runs = []\n",
+    "    create_run_latencies = []\n",
+    "    for i in range(num_runs):\n",
+    "        resource_references = []\n",
+    "        key = kfp_server_api.models.ApiResourceKey(id=experiment_id, type=kfp_server_api.models.ApiResourceType.EXPERIMENT)\n",
+    "        reference = kfp_server_api.models.ApiResourceReference(key=key, relationship=kfp_server_api.models.ApiRelationship.OWNER)\n",
+    "        resource_references.append(reference)\n",
+    "        key = kfp_server_api.models.ApiResourceKey(id=default_version_id, type=kfp_server_api.models.ApiResourceType.PIPELINE_VERSION)\n",
+    "        reference = kfp_server_api.models.ApiResourceReference(key=key, relationship=kfp_server_api.models.ApiRelationship.CREATOR)\n",
+    "        resource_references.append(reference)\n",
+    "        # If the pipeline you choose needs to specify parameters to create a run, specify it here.\n",
+    "        parameters = []\n",
+    "        parameter = kfp_server_api.ApiParameter(name='pipeline-root', value='gs://jingzhangjz-project-outputs/tfx_taxi_simple/{{workflow.uid}}')\n",
+    "        parameters.append(parameter)\n",
+    "        parameter = kfp_server_api.ApiParameter(name='data-root', value='gs://ml-pipeline-playground/tfx_taxi_simple/data')\n",
+    "        parameters.append(parameter)\n",
+    "        parameter = kfp_server_api.ApiParameter(name='module-file', value='gs://ml-pipeline-playground/tfx_taxi_simple/modules/taxi_utils.py')\n",
+    "        parameters.append(parameter)        \n",
+    "        pipeline_spec = kfp_server_api.ApiPipelineSpec(parameters=parameters)\n",
+    "\n",
+    "        start = time.perf_counter()\n",
+    "        run_name = 'run-' + random_suffix()\n",
+    "        run = client.runs.create_run(body={'name':run_name, 'resource_references': resource_references, 'pipeline_spec': pipeline_spec})  \n",
+    "        dur = time.perf_counter() - start\n",
+    "        create_run_latencies.append(dur)   \n",
+    "        created_runs.append(run.run.id)\n",
+    "               \n",
+    "    # Wait for the runs to finish.\n",
+    "    while True:\n",
```

Comment:
nit: shall we add a timeout in case it is stuck?
not blocking, you can add a TODO item here

Reply:
added todo.